pkg/services/ngalert/remote/alertmanager.go

```diff
@@ -647,6 +647,16 @@ func (am *Alertmanager) Ready() bool {
 	return am.ready
 }
 
+// It makes no sense for us to call these methods on the remote Alertmanager implementation.
+// TODO: Remove them, let the internal and remote Alertmanagers have different APIs.
+func (am *Alertmanager) MergeNflog(b []byte) error {
+	panic("Can't merge nflog on the remote Alertmanager")
+}
+
+func (am *Alertmanager) MergeSilences(b []byte) error {
+	panic("Can't merge silences on the remote Alertmanager")
+}
```

Comment:
Making the remote and internal Alertmanager APIs different will require a big refactor.
This should be enough to prevent any errors on our side (calling these methods on the wrong Alertmanager).